### ISSUE TITLE
Turbopack & Pages Router: Winston TypeError: __TURBOPACK__commonjs__external__winston__.default.createLogger is not a function

### ISSUE THREAD

**Issue by ivanfeli** [vercel/next.js#63795]
### Link to the code that reproduces this issue

https://github.com/ivanfeli/turbo-winston-repro-pages

### To Reproduce

**Steps:**
* pnpm dev
* go to localhost:PORT/
* crash

_lib/winston.ts_
```ts
import winston from 'winston'

export const logger = winston.createLogger({
    level: 'info',
})
```

_pages/index.tsx_
```
import {logger} from "../lib/winston";

export default function Home() {
  return <h1>Hi</h1>;
}

export const getServerSideProps = async () => {
    // crashes here
    logger.info('Hello from getServerSideProps');
    return { props: {} };
}
```

### Current vs. Expected behavior

The following error is seen in the browser and console:

```
⨯ TypeError: __TURBOPACK__commonjs__external__winston__.default.createLogger is not a function
    at C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\_77a029._.js:12:70
    at [project]/lib/winston.ts [ssr] (ecmascript) (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\_77a029._.js:16:3)
    at instantiateModule (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:511:23)
    at getOrInstantiateModuleFromParent (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:563:12)
    at esmImport (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:113:20)
    at C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\_77a029._.js:25:111
    at [project]/pages/index.tsx [ssr] (ecmascript) (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\_77a029._.js:46:3)
    at instantiateModule (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:511:23)
    at getOrInstantiateModuleFromParent (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:563:12)
    at esmImport (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:113:20)
    at C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\node_modules__pnpm_abc74f._.js:2492:112
    at [project]/node_modules/.pnpm/next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/build/templates/pages.js { INNER_PAGE => "[project]/pages/index.tsx [ssr] (ecmascript)", INNER_DOCUMENT => "[proj
ect]/node_modules/.pnpm/next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0/node_modules/next/document.js [ssr] (ecmascript)", INNER_APP => "[project]/node_modules/.pnpm/next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0/node_m
odules/next/app.js [ssr] (ecmascript)" } [ssr] (ecmascript) (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\node_modules__pnpm_abc74f._.js:2527:3)
    at instantiateModule (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:511:23)
    at instantiateRuntimeModule (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:571:12)
    at Object.getOrInstantiateRuntimeModule (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\chunks\ssr\[turbopack]_runtime.js:586:12)
    at Object.<anonymous> (C:\Users\xxx\src\turbo-winston-repro-pages\.next\server\pages\index.js:5:26)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at mod.require (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\require-hook.js:65:28)
    at require (node:internal/modules/cjs/helpers:119:18)
    at requirePage (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\require.js:109:84)
    at C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\load-components.js:98:84
    at async loadComponentsImpl (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\load-components.js:98:26)
    at async DevServer.findPageComponentsImpl (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\next-server.js:682:36)
    at async DevServer.findPageComponents (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\dev\next-dev-server.js:577:20)
    at async DevServer.renderPageComponent (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1899:24)
    at async DevServer.renderToResponseImpl (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1951:32)
    at async DevServer.pipeImpl (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:911:25)
    at async NextNodeServer.handleCatchallRenderRequest (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\next-server.js:272:17)    
    at async C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\dev\next-dev-server.js:339:20
    at async Span.traceAsyncFn (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\trace\trace.js:154:20)
    at async DevServer.handleRequest (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\dev\next-dev-server.js:336:24)
    at async invokeRender (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\router-server.js:174:21)
    at async handleRequest (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\router-server.js:353:24)
    at async requestHandlerImpl (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\router-server.js:377:13)
    at async Server.requestListener (C:\Users\xxx\src\turbo-winston-repro-pages\node_modules\.pnpm\next@14.2.0-canary.46_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\start-server.js:141:13) {
  page: '/'
}
```

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
  Available memory (MB): 32717
  Available CPU cores: 12
Binaries:
  Node: 18.18.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.0-canary.46 // Latest available version is detected (14.2.0-canary.46).
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack (--turbo)

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

As per my limited testing, it seems to work fine in the App Router.

<sub>[PACK-2878](https://linear.app/vercel/issue/PACK-2878/turbopack-and-pages-router-winston-typeerror-turbopack-commonjs)</sub>

**Comment by ivanfeli** [comment]
FYI; still happening in canary 64

**Comment by ardokirsipuu** [comment]
Ahh, thanks for the additional context! Our project had only a few things left under /pages/api/ and after moving them to /app/api/ the error went away. 🎉

Also, I couldn't find any mention of this at https://nextjs.org/docs/architecture/turbopack#unsupported-features

**Comment by ivanfeli** [comment]
This seems to have been fixed as of `next@v15.0.0-canary.42`
